components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/NetworkMessageEncoder.cs

```diff
@@ -101,6 +102,7 @@ public class NetworkMessageEncoder : IMessageEncoder {
                     MessageContentMask = message.WriterGroup.MessageSettings.NetworkMessageContentMask
                         .ToStackType(message.WriterGroup?.MessageType),
                     PublisherId = message.PublisherId,
+                    DataSetClassId = message.Writer?.DataSet?.DataSetMetaData?.DataSetClassId.ToString(),
```

Comment:
Should we use AsString() if this is a NodeId, or if it is a string, can we remove?

Reply:
DataSetClassId is a guid